sonar-scanner-engine/src/main/java/org/sonar/scanner/cpd/deprecated/CpdBlockIndexer.java

```diff
@@ -19,31 +19,29 @@
  */
 package org.sonar.scanner.cpd.deprecated;
 
-import org.slf4j.Logger;
 import org.sonar.api.batch.ScannerSide;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
 
 @ScannerSide
 public abstract class CpdBlockIndexer {
 
+  private static final Logger LOG = Loggers.get(CpdBlockIndexer.class);
+
   abstract boolean isLanguageSupported(String language);
 
   abstract void index(String language);
 
-  protected void logExclusions(String[] exclusions, Logger logger) {
+  protected void logExclusions(String[] exclusions) {
     if (exclusions.length > 0) {
       StringBuilder message = new StringBuilder("Copy-paste detection exclusions:");
       for (String exclusion : exclusions) {
         message.append("\n  ");
         message.append(exclusion);
       }
 
-      logger.info(message.toString());
+      LOG.info("{}", message);
```

Comment:
message is a StringBuilder. If I don't wan't to pay the cost of "building" the final String, that's the only way I have found